File: Servidor/models/product.js
'use strict';

const {Model} = require('sequelize');

// Definition of the Quiz model:
module.exports = (sequelize, DataTypes) => {
    class Product extends Model {
    }

    Product.init(
        {
          reference: {
              type: DataTypes.STRING
          },
          color: {
              type: DataTypes.STRING
          },
          category: {
              type: DataTypes.STRING
          },
          price: {
              type: DataTypes.DOUBLE
          },
          composition: {
              type: DataTypes.STRING
          }
        }, 
        { sequelize }
      );

    return Product;
};

File: Approach/Approach/Model/Entities/Product.swift
//
//  Product.swift
//  Approach
//
//  Created by Javier Mazario on 16/9/21.
//

import Foundation
import SwiftUI

struct Product: Decodable, Identifiable {
    var id: Int
    var productId: String
    var reference: String
    var color: String
    var category: String
    var price: Double
    var composition: String
    var isAvailable: Bool
    let imageInfo: ImageFile?
    
    struct ImageFile: Codable {
        let filename: String?
        let mime: String?
        let url: URL?
    }
}

extension Product {
    var image: Image {
        Image(uiImage: ImageStore.shared.image(url: imageInfo!.url!))
    }
}
